pkg/virt-controller/watch/migration.go

```diff
@@ -843,7 +846,11 @@ func (c *MigrationController) handleTargetPodCreation(key string, migration *vir
 				return nil
 			}
 		}
-		return c.createTargetPod(migration, vmi, sourcePod)
+		backendStoragePVC, err := backendstorage.CreateIfNeeded(vmi, c.clusterConfig, c.clientset, backendstorage.IsMigration)
```

Comment:
I'm a bit lost. Could you please explain why do we need to do this?
I mean why can't we simply skip this volume during migration and re-use the original on the destination? ( same as we do with other volumes)

Reply:
That would be nice but I don't think we can.
When talking about volumes used by the VM it's not a problem, since the VM only ever runs in one place.
However, in this case, the contents of the PVC would be in use in 2 places at once (source container and target container).
In the case of swtpm, we could end up with corrupt data. In fact, swtpm takes a file-based lock to prevent exactly that, so the migration actually fails when using the original in the destination.

Reply:
> However, in this case, the contents of the PVC would be in use in 2 places at once (source container and target container).

How can this be in use in two places? During the migration, the target domain is paused and will only be unpaused when the source is dead.
Why would this case be any different?

> In the case of swtpm, we could end up with corrupt data. In fact, swtpm takes a file-based lock to prevent exactly that, so the migration actually fails when using the original in the destination.
We would end up in this situation for any migration, but this is not happening because of what I've described above. Also, source qemu normally locks so the destination won't write into it accidentally.

Reply:
> > However, in this case, the contents of the PVC would be in use in 2 places at once (source container and target container).
> 
> How can this be in use in two places? During the migration, the target domain is paused and will only be unpaused when the source is dead. Why would this case be any different?

Yes, the target domain is paused but not the target pod! This case is different because the contents of the PVC are actually accessed by the pod itself, not the domain.

> 
> > In the case of swtpm, we could end up with corrupt data. In fact, swtpm takes a file-based lock to prevent exactly that, so the migration actually fails when using the original in the destination.
> 
> We would end up in this situation for any migration, but this is not happening because of what I've described above. Also, source qemu normally locks so the destination won't write into it accidentally.

Yup, this again applies to the domain but not to its hosting pod.

Reply:
> > > However, in this case, the contents of the PVC would be in use in 2 places at once (source container and target container).
> > 
> > 
> > How can this be in use in two places? During the migration, the target domain is paused and will only be unpaused when the source is dead. Why would this case be any different?
> 
> Yes, the target domain is paused but not the target pod! This case is different because the contents of the PVC are actually accessed by the pod itself, not the domain.

Can you please explain why the pod is accessing this content (especially if the domain is not running) if this content belongs to the domain?

Filesystem volumes that we attach to the VMIs should behave similarly. These volumes are mounted in the pod (compute container) and have a diskX.img file that is being used by qemu. 
It seems to me like a similar scenario. If not, can you please explain why not?

Reply:
Ok I see your point now... Even when coming from the domain, most I/O operations ultimately happen in the backend. I guess a more accurate answer to your question is that qemu is equipped to handle cases where the same volume is mounted on both the source and the target, and swtpm is not... We could open an issue against swtpm and switch to using just 1 PVC if/when the issue gets resolved.

Reply:
> Ok I see your point now... Even when coming from the domain, most I/O operations ultimately happen in the backend. I guess a more accurate answer to your question is that qemu is equipped to handle cases where the same volume is mounted on both the source and the target, and swtpm is not... We could open an issue against swtpm and switch to using just 1 PVC if/when the issue gets resolved.

Qemu is able to handle this mainly because the domain is paused and will not be unpaused until the source is destroyed. The same should happen with swptm. What is the reason this service would generate any I/O if qemu, which is its only consumer, is paused?

Reply:
Yes, great question. It almost feels like that lock swtpm has in place is there to block this very scenario, I think we should ask about it, I'll open an issue.

Reply:
I'm not sure which lock would that be, but if this code has been accepted into libvirt then Im sure this volume should be treated as any other. 

Honestly, I'd be against us developing an additional mechanism to handle PVCs , instead this volume should be treated the same way as any other filesystem volume we attach to the guests.

Reply:
This lock (inside virt-launcher/compute, running as non-root): /var/run/kubevirt-private/libvirt/qemu/swtpm/09d086e6-ba5d-4647-8f7d-c16d17ae20c0/tpm2/.lock

Please keep in mind that we do not attach this volume to the guest! Libvirt is not aware of it, and the libvirt TPM option do not have a way to provide a specific disk or even location for storing the TPM state.